src/modules/samples/cpp/src/lib/Sample.cpp

```diff
@@ -0,0 +1,165 @@
+
+OSCONFIG_LOG_HANDLE SampleLog::m_log = nullptr;
+
+Sample::Sample(unsigned int maxPayloadSizeBytes)
+{
+    m_maxPayloadSizeBytes = maxPayloadSizeBytes;
+}
+
+int Sample::Set(const char* componentName, const char* objectName, const MMI_JSON_STRING payload, const int payloadSizeBytes)
+{
+    int status = MMI_OK;
+    rapidjson::Document document;
+
+    if (document.Parse(payload, payloadSizeBytes).HasParseError())
+    {
+        OsConfigLogError(SampleLog::Get(), "Unabled to parse JSON payload: %s", payload);
+        status = EINVAL;
+    }
+    else
+    {
+        // Dispatch the request to the appropriate method for the given component and object
+        if (0 == g_componentName.compare(componentName))
+        {
+            if (0 == g_objectName.compare(objectName))
+            {
+                // Get the required data from the payload and dispatch the request to the client session
+                if (document.IsString())
+                {
+                    // Apply the payload data
+                    m_value = document.GetString();
+                    status = MMI_OK;
+                }
+                else
+                {
+                    OsConfigLogError(SampleLog::Get(), "JSON payload is not a string");
+                    status = EINVAL;
+                }
+            }
+            else
+            {
+                OsConfigLogError(SampleLog::Get(), "Invalid objectName: %s", objectName);
+                status = EINVAL;
+            }
+        }
+        else
+        {
+            OsConfigLogError(SampleLog::Get(), "Invalid componentName: %s", componentName);
+            status = EINVAL;
+        }
+    }
+
+    return status;
+}
+
+int Sample::Get(const char* componentName, const char* objectName, MMI_JSON_STRING* payload, int* payloadSizeBytes)
+{
+    int status = MMI_OK;
+
+    if (nullptr == payloadSizeBytes)
+    {
+        OsConfigLogError(SampleLog::Get(), "Invalid payloadSizeBytes");
+        return EINVAL;
+    }
+
+    *payload = nullptr;
+    *payloadSizeBytes = 0;
+
+    unsigned int maxPayloadSizeBytes = GetMaxPayloadSizeBytes();
+    rapidjson::Document document;
+
+    // Dispatch the get request to the appropriate method for the given component and object
+    if (0 == g_componentName.compare(componentName))
+    {
+        if (0 == g_objectName.compare(objectName))
+        {
+            std::string value = m_value;
+            document.SetString(value.c_str(), document.GetAllocator());
+
+            // Serialize the JSON object to the payload buffer
+            status = Sample::SerializeJsonPayload(document, payload, payloadSizeBytes, maxPayloadSizeBytes);
+        }
+        else
+        {
+            OsConfigLogError(SampleLog::Get(), "Invalid objectName: %s", objectName);
+            status = EINVAL;
+        }
+    }
+    else
+    {
+        OsConfigLogError(SampleLog::Get(), "Invalid componentName: %s", componentName);
```

Comment:
>OsConfigLogError(SampleLog::Get(), "Invalid componentName: %s", componentName); [](http://example.com/codeflow?start=8&length=79)

Sample should showcase how to limit certain logging (that can become repetitive and fill out the logs with same lines ober and over) using IsFullLoggingEnabled()

Reply:
Do you think it is valuable to add these checks to `Sample.cpp` too? The functionality of `IsFullLogging()` is showcased in the `ScopeGuards` of `SampleModule.cpp`

Reply:
We want to highlight this is a concern for MmiSet and MmiGet for traces in general.

Reply:
Looks good as this for simplicity. Thanks!